core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -558,7 +567,36 @@ public Result visit(Aggregate e) {
         return result(select, ImmutableList.of(Clause.SELECT), e, null);
       }
     }
-    return builder.result();
+    Result result = builder.result();
+    return updateResult(e, result);
+  }
+
+  Result updateResult(Aggregate e, Result x) {
+    // CTE Scope Trait
```

Comment:
i think this is redundant code used in multiple methods, it can be moved to just one.

Reply:
Done